mdp/nodes/misc_nodes.py

```diff
@@ -815,3 +826,603 @@ def _execute(self, x):
+                rimg.append(mdp.numx.array(imx.resize(size_xy[::-1])))
+        else:
+            raise mdp.NodeException("OpenCV python bindings or PIL Image package is required to resize images.")
+
+        return mdp.numx.asarray(rimg)
+
+    @staticmethod
+    def _img_255_1(x):
+        """Scales uint img data to float values between [0, 1]"""
+        x = old_div(x, 255.)
+        return x
+
+    @staticmethod
+    def _gray(x):
+        """Converts the RGB image to grayscale.
+        The last dimension is expected to represent the color channel
+        """
+        return x.mean(axis=-1)
+
+    def _to_2d(self, x):
+        """Reshapes data to 2D.
+        Also, stores the input shape information in 'self._shape_buffer'.
+        This can used to later restore the data's shape.
+        """
+        if x.ndim == 2:
+            return x
+        self._shape_buffer = x.shape[1:]
+        return x.reshape(x.shape[0], mdp.numx.product(x.shape[1:]))
+
+    @staticmethod
+    def _to_dtype(x, dtype):
+        """Changes the data type of the input.
+        Useful when dealing with Image data.
+        """
+        return x.astype(dtype)
+
+    def _set_shape(self, x, shape=None):
+        """Reshapes the input data to the given shape.
+        Useful for processing images. If given shape is None,
+        the stored self._shape_buffer is used instead.
+        """
+        print self._shape_buffer
+        if shape is None:
+            # check if it can be reshapes with the stored shape buffer
+            if mdp.numx.product(self._shape_buffer) == mdp.numx.product(x.shape[1:]):
+                return x.reshape(x.shape[0], *self._shape_buffer)
+            else:
+                raise mdp.NodeException("Cannot reshape to the stored shape buffer. Provide the desired shape.")
+        else:
+            return x.reshape(x.shape[0], *shape)
+
+    # Node methods
+
+    @staticmethod
+    def is_trainable():
+        return False
+
+    @staticmethod
+    def is_invertible():
+        return False
+
+    def _execute(self, x):
+        if self.transform_seq is None:
+            # no transformations
+            return x
+        elif self.transform_seq_args is None:
+            # checks if no args needed
+            self.transform_seq_args = [None] * len(self.transform_seq)
+
+        _transform_fns = self._get_transform_fns()
+        for i, fn_name in enumerate(self.transform_seq):
+            x = _transform_fns[fn_name](x, *self.transform_seq_args[i])
+
+        # reset shape buffer
+        self._shape_buffer = self.input_shape
+
+        # output to 2d
+        x = x.reshape(x.shape[0], mdp.numx.product(x.shape[1:]))
+
+        return self._refcast(x)
+
+    # utility methods
+
+    def get_transform_fn_names(self):
+        """Returns the available transformation function names"""
+        fns = self._get_transform_fns()
+        return fns.keys()
+
+
+# TODO: Could use or add to the graph package of MDP?
```

Comment:
Thanks for pointing this out. I will update the todo and will take a look into it.